proposals/003-metric-improvenents.md

```diff
@@ -0,0 +1,262 @@
+| kroxylicious_downstream_connections_total     | Counter | virtual_cluster, node_id†  | virtualCluster    | Incremented by one every time a request comes from/response goes to the downstream. |
+| kroxylicious_downstream_error_total           | Counter | virtual_cluster†, node_id† | virtualCluster    | Incremented by one every time a request goes to/response comes from the upstream.   |
+| kroxylicious_upstreamstream_connections_total | Counter | virtual_cluster,  node_id† | virtualCluster    | Incremented by one every time a request comes from/response goes to the downstream. |
+| kroxylicious_upstreamstream_error_total       | Counter | virtual_cluster,  node_id† | virtualCluster    | Records the time taken for the message to traverse the proxy                        |
+
+
+The following metrics will be deprecated.
+
+| Metric                                           |
+|--------------------------------------------------|
+| kroxylicious_inbound_downstream_messages         |
+| kroxylicious_inbound_downstream_decoded_messages |
+| kroxylicious_payload_size_bytes                  |
+
+
+(† signifies labels that will be omitted in some circumstances. The circumstances are explained below).
+
+### New counter `kroxylicious_downstream_kafka_message_total`
+
+`kroxylicious_downstream_kafka_message_total` is incremented every time a message passes to/from the downstream.
+It is discriminated by `virtual_cluster`, `node_id`, `flowing`, `opaque`, api_key`, and `api_version` labels.
+
+`node_id` will only be present if for requests going to or responses coming from a broker.  For bootstrap interactions,
+`node_id` will be absent.
+
+`flowing` has two values - `upstream` signifies a request following towards the broker, `downstream` signifies a response
+following towards the client.
+
+`opaque` has two values - `1` signifies an opaque message (one that the proxy hasn't needed to decode), `0` signifies decoded message.
+
+Use cases that were previously served by the deprecated metrics `kroxylicious_inbound_downstream_messages` and
+`kroxylicious_inbound_downstream_decoded_messages` can now use this metric instead.
+
+### New counter `kroxylicious_upstream_kafka_message_total`
+
+`kroxylicious_upstream_kafka_message_total` is incremented every time a message passes to/from the upstream.
+It is discriminated by `virtual_cluster`, `node_id`, `flowing`, `opaque`, `api_key`, `api_version` and `originating_filter` labels.
+
+`node_id` will only be present if for requests going to or responses coming from a broker.  For bootstrap interactions,
+`node_id` will be absent.
+
+`originating_filter` is only used for requests generated by the filters themselves (`sendRequest` API)
+and their responses.  It will be populated with the filter's name. Otherwise `originating_filter` will be absent. `opaque`
```

Comment:
I know in the API we use the term opaque. But we should remember that people operating the proxy are not going to know that terminology. Using `decoded` would be more self-explanatory.